lambda/matching.ml

```diff
@@ -290,25 +358,25 @@ end = struct
             Ident.create_local (Ident.name id) ))
       ids
 
-  let explode_or_pat (p, patl) ~arg ~mk_action ~vars rem =
+  let explode_or_pat ((p : Half_simple.pattern), patl) ~arg ~mk_action ~vars
+      (rem : clause list) : clause list =
     let rec explode p aliases rem =
+      let split_explode p aliases rem = explode (General.view p) aliases rem in
       match p.pat_desc with
-      | Tpat_or (p1, p2, _) -> explode p1 aliases (explode p2 aliases rem)
-      | Tpat_alias (p, id, _) -> explode p (id :: aliases) rem
-      | Tpat_var (x, _) ->
-          let env = mk_alpha_env arg (x :: aliases) vars in
-          ((omega, patl), mk_action ~vars:(List.map snd env)) :: rem
-      | _ ->
+      | `Or (p1, p2, _) ->
+          split_explode p1 aliases (split_explode p2 aliases rem)
+      | `Alias (p, id, _) -> split_explode p (id :: aliases) rem
+      | `Var (id, str) ->
+          explode
+            { p with pat_desc = `Alias (Parmatch.omega, id, str) }
+            aliases rem
```

Comment:
The behavior is the same as before this PR, but the previous version inlined the recursive call to `explode` whereas the new code goes through an `` `Alias `` intermediary stage. I imagine that this is done to follow the code path in `Half_simplified.of_clause`?

Reply:
The idea was that it was clearer to just rewrite `x` into `_ as x` on the fly than to copy the implementation of the Alias and omega cases in the Var case.